cpp/src/arrow/compute/exec/test_util.cc

```diff
@@ -459,5 +460,44 @@ void PrintTo(const Declaration& decl, std::ostream* os) {
   *os << "}";
 }
 
+std::shared_ptr<Table> MakeRandomTable(TableGenerationProperties properties) {
```

Comment:
Can you rename this table so it is obvious it has something to do with asof join / time series data?  We have a couple of different random generation utilities already and so it will help to distinguish.